modules/3d-tiles/src/3d-tiles-archive/3d-tiles-archive-archive.ts

```diff
@@ -22,19 +22,17 @@ const COMPRESSION_METHODS: {[key: number]: CompressionHandler} = {
 /**
  * Class for handling information about 3tz file
  */
-export class Tiles3DArchive {
-  /** FileProvider with whe whole file */
-  private fileProvider: FileProvider;
+export class Tiles3DArchive extends IndexedArchive {
```

Comment:
Just a question, is inheritance the best pattern here? Could this instead have a member `indexedArchive`.

I am not familiar enough with the design to have a strong opinion, but object oriented subclassing is a fairly strong level of coupling, and it is sometimes used be default when not strictly necessary.

Indicators that subclass is the right approach would be that there is code that works on (accepts as parameters etc) `IndexedArchives` that now can work with a number of `IndexedArchive` subclasses.

Reply:
The super class IndexedArchive is used in ZipFileSystem. It allows to unbind from specific formats (SLPK/3TZ)